components/eventing-controller/controllers/subscriptionv2/jetstream/reconciler.go

```diff
@@ -197,81 +184,121 @@ func (r *Reconciler) enqueueReconciliationForSubscriptions(subs []eventingv1alph
 	}
 }
 
-// syncSubscriptionStatus syncs Subscription status and keeps the status up to date.
-func (r *Reconciler) syncSubscriptionStatus(ctx context.Context,
-	sub *eventingv1alpha2.Subscription, updateStatus bool, err error) error {
-	isNatsReady := err == nil
-	readyStatusChanged := setSubReadyStatus(&sub.Status, isNatsReady)
-
-	desiredConditions := initializeDesiredConditions()
-	setConditionSubscriptionActive(desiredConditions, err)
-	// check if the conditions are missing or changed
-	if !eventingv1alpha2.ConditionsEquals(sub.Status.Conditions, desiredConditions) {
-		sub.Status.Conditions = desiredConditions
-		updateStatus = true
-	}
-
-	// Update the status only if something needs to be updated
-	if updateStatus || readyStatusChanged {
-		if updateErr := r.Client.Status().Update(ctx, sub, &client.UpdateOptions{}); updateErr != nil {
-			events.Warn(r.recorder, sub, events.ReasonUpdateFailed, "Update Subscription status failed %s", sub.Name)
-			return pkgerrors.MakeError(errFailedToUpdateStatus, updateErr)
-		}
-		events.Normal(r.recorder, sub, events.ReasonUpdate, "Update Subscription status succeeded %s", sub.Name)
-	}
-	return nil
-}
-
 // handleSubscriptionDeletion deletes the JetStream subscription and removes its finalizer if it is set.
 func (r *Reconciler) handleSubscriptionDeletion(ctx context.Context,
-	subscription *eventingv1alpha2.Subscription, log *zap.SugaredLogger) error {
+	subscription *eventingv1alpha2.Subscription, log *zap.SugaredLogger) (ctrl.Result, error) {
+	// delete the JetStream subscription/consumer
 	if utils.ContainsString(subscription.ObjectMeta.Finalizers, eventingv1alpha2.Finalizer) {
 		if err := r.Backend.DeleteSubscription(subscription); err != nil {
 			// if failed to delete the external dependency here, return with error
 			// so that it can be retried
-			return pkgerrors.MakeError(errFailedToDeleteSub, err)
+			if syncErr := r.syncSubscriptionStatus(ctx, subscription, err, log); syncErr != nil {
+				return ctrl.Result{}, syncErr
+			}
+			return ctrl.Result{}, pkgerrors.MakeError(errFailedToDeleteSub, err)
 		}
 
-		// remove our finalizer from the list and update it.
+		// remove the eventing finalizer from the list and update the subscription.
 		subscription.ObjectMeta.Finalizers = utils.RemoveString(subscription.ObjectMeta.Finalizers,
 			eventingv1alpha2.Finalizer)
-		if err := r.Client.Update(ctx, subscription); err != nil {
-			events.Warn(r.recorder, subscription, events.ReasonUpdateFailed, "Update Subscription failed %s", subscription.Name)
-			return pkgerrors.MakeError(errFailedToRemoveFinalizer, err)
+		return ctrl.Result{}, r.syncSubscriptionStatus(ctx, subscription, nil, log)
```

Comment:
I don't see why the status needs updating when I read this function. I know that the functionality is cascaded down. I would like to have seen something like:
```suggestion
		return ctrl.Result{}, r.syncSubscription(ctx, subscription)
```

Reply:
I refactored the code, check again. I only update the subscription now, no status changes there.